operate/schema/src/main/java/io/camunda/operate/schema/IndexSchemaValidator.java

```diff
@@ -135,4 +147,92 @@ public boolean schemaExists() {
+   *
+   * @return newFields map with the new field definitions per index
+   * @throws OperateRuntimeException in case some fields would need to be deleted or have different
+   *     settings
+   */
+  public Map<IndexDescriptor, Set<IndexMappingProperty>> validateIndexMappings() {
+    Map<IndexDescriptor, Set<IndexMappingProperty>> newFields = new HashMap<>();
+    // we only check main indices considering that archived have the same structure
+    // the necessary updates will be applied to both main and archived indices
+    final Map<String, IndexMapping> indexMappings =
+        schemaManager.getIndexMappings(schemaManager.getIndexPrefix() + "*");
+    for (IndexDescriptor indexDescriptor : indexDescriptors) {
+      Map<String, IndexMapping> indexMappingsGroup =
+          filterIndexMappings(indexMappings, indexDescriptor);
+      // we don't check indices that were not yet created
+      if (!indexMappingsGroup.isEmpty()) {
+
+        IndexMapping indexMappingMustBe = schemaManager.getExpectedIndexFields(indexDescriptor);
+        IndexMappingDifference difference = null;
+        // compare every index in group
+        for (Map.Entry<String, IndexMapping> singleIndexMapping : indexMappingsGroup.entrySet()) {
+          final IndexMappingDifference currentDifference =
+              indexMappingMustBe.difference(singleIndexMapping.getValue());
+          if (!currentDifference.isEqual()) {
+            if (difference == null) {
+              difference = currentDifference;
+            } else if (!difference.equals(currentDifference)) {
+              throw new OperateRuntimeException(
+                  "Ambiguous schema update. First bring runtime and date indices to one schema. Difference 1: "
+                      + difference
+                      + ". Difference 2: "
+                      + currentDifference);
+            }
+          }
+        }
+        if (difference != null && !difference.isEqual()) {
+          logger.debug(
+              String.format(
+                  "Index fields differ from expected. Index name: %s. Difference: %s.",
+                  indexDescriptor.getIndexName(), difference));
+          if (!difference.getEntriesDiffering().isEmpty()) {
+            final String errorMsg =
+                String.format(
+                    "Index name: %s. Not supported index changes are introduced. Data migration is required. Changes found: %s",
+                    indexDescriptor.getIndexName(), difference.getEntriesDiffering());
+            logger.error(errorMsg);
+            throw new OperateRuntimeException(errorMsg);
+          } else if (!difference.getEntriesOnlyOnRight().isEmpty()) {
+            final String message =
+                String.format(
+                    "Index name: %s. Field deletion is requested, will be ignored. Fields: %s",
+                    indexDescriptor.getIndexName(), difference.getEntriesOnlyOnRight());
+            logger.warn(message);
```

Comment:
Should this be logged as warning if we consider it expected behavior in rolling update scenarios? Warning will let people believe that something is wrong, and the message doesn't help them resolve it. So I'd suggest to go either way: Choose another log level or give more information what this message means and what to do with it.